### PR TITLE
statistics: add `last_analyze_version` for `mysql.stats_meta`

### PR DESCRIPTION
**Pull request by winoros**
### What problem does this PR solve?

Issue Number: ref https://github.com/pingcap/tidb/issues/49594

Problem Summary:

Add one more column for `mysql.stats_meta` for usability.

### What changed and how does it work?

- We don't update the value in TiKV when we upgrade TiDB.
- Update the kv value when analyzing it.

We haven't changed the table `mysql.stats_meta_history` yet.

### Check List

Tests 

- [x] Unit test
- [x] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Comment by winoros** [comment]
https://github.com/pingcap/tidb/pull/47496 cleaned the memory usage of those tables that have no SELECT statements on the TiDB node. So we need this pull to tell the autoanalyze worker the last_analyze_time of those tables.

File: br/pkg/restore/db_test.go [hunk 1]
```diff
@@ -425,5 +425,5 @@ func TestGetExistedUserDBs(t *testing.T) {
 //
 // The above variables are in the file br/pkg/restore/systable_restore.go
 func TestMonitorTheSystemTableIncremental(t *testing.T) {
-	require.Equal(t, int64(185), session.CurrentBootstrapVersion)
+	require.Equal(t, int64(186), session.CurrentBootstrapVersion)
```
Comment (lance6716) [inline br/pkg/restore/db_test.go:428]: /cc @Leavrth please check if BR needs to make more change

Comment (Leavrth) [reply br/pkg/restore/db_test.go:428]: Is there any risk to directly update `last_analyze_version` with `restoreTS`?
https://github.com/pingcap/tidb/blob/dd1c635a8bc22a4220f402b7401f62e0cad4c553/br/pkg/restore/db.go#L141-L160

File: pkg/executor/show_stats.go [hunk 2]
```diff
@@ -141,14 +141,27 @@ func (e *ShowExec) appendTableForStatsMeta(dbName, tblName, partitionName string
...
 	}
-	e.appendRow([]any{
-		dbName,
-		tblName,
-		partitionName,
-		e.versionToTime(statsTbl.Version),
-		statsTbl.ModifyCount,
-		statsTbl.RealtimeCount,
-	})
+	if statsTbl.LastAnalyzeVersion == 0 {
```
Comment (hi-rustin) [inline pkg/executor/show_stats.go:144]: I remember we have a statsxx.Version0 const. Maybe we can use it here.

File: pkg/executor/test/analyzetest/analyze_test.go [hunk 3]
```diff
@@ -1394,9 +1394,9 @@ func TestAnalyzeColumnsWithDynamicPartitionTable(t *testing.T) {
 
 			rows = tk.MustQuery("show stats_meta where db_name = 'test' and table_name = 't'").Sort().Rows()
 			require.Equal(t, 3, len(rows))
-			require.Equal(t, []any{"test", "t", "global", "0", "20"}, append(rows[0][:3], rows[0][4:]...))
-			require.Equal(t, []any{"test", "t", "p0", "0", "9"}, append(rows[1][:3], rows[1][4:]...))
-			require.Equal(t, []any{"test", "t", "p1", "0", "11"}, append(rows[2][:3], rows[2][4:]...))
+			require.Equal(t, []any{"test", "t", "global", "0", "20"}, append(rows[0][:3], rows[0][4:6]...))
+			require.Equal(t, []any{"test", "t", "p0", "0", "9"}, append(rows[1][:3], rows[1][4:6]...))
+			require.Equal(t, []any{"test", "t", "p1", "0", "11"}, append(rows[2][:3], rows[2][4:6]...))
```
Comment (hi-rustin) [inline pkg/executor/test/analyzetest/analyze_test.go:1399]: Can we add a case to cover the LastAnalyzeVersion?

File: pkg/statistics/handle/autoanalyze/autoanalyze.go [hunk 4]
```diff
@@ -516,7 +516,7 @@ func tryAutoAnalyzeTable(
 //
 // Exposed for test.
 func NeedAnalyzeTable(tbl *statistics.Table, autoAnalyzeRatio float64) (bool, string) {
-	analyzed := exec.TableAnalyzed(tbl)
```
Comment (hi-rustin) [inline pkg/statistics/handle/autoanalyze/autoanalyze.go:519]: I guess we can add a tbl.IsAnalyzed API to simplify this code? Then we don't need to write this check everywhere.

File: pkg/statistics/handle/autoanalyze/refresher/refresher.go [hunk 5]
```diff
@@ -370,7 +370,7 @@ func CalculateChangePercentage(
 	tblStats *statistics.Table,
 	autoAnalyzeRatio float64,
 ) float64 {
-	if !exec.TableAnalyzed(tblStats) {
+	if tblStats.LastAnalyzeVersion == 0 {
```
Comment (hi-rustin) [inline pkg/statistics/handle/autoanalyze/refresher/refresher.go:373]: ditto

File: pkg/statistics/handle/autoanalyze/refresher/refresher.go [hunk 6]
```diff
@@ -424,23 +424,12 @@ func findLastAnalyzeTime(
...
-		}
-	}
-	for _, col := range tblStats.Columns {
-		if col.IsAnalyzed() {
-			maxVersion = max(maxVersion, col.LastUpdateVersion)
-		}
-	}
 	// Table is not analyzed, compose a fake version.
-	if maxVersion == 0 {
+	if tblStats.LastAnalyzeVersion == 0 {
```
Comment (hi-rustin) [inline pkg/statistics/handle/autoanalyze/refresher/refresher.go:428]: ditto

File: pkg/statistics/handle/autoanalyze/refresher/refresher.go [hunk 7]
```diff
@@ -450,7 +439,7 @@ func CheckIndexesNeedAnalyze(
 ) []string {
 	// If table is not analyzed, we need to analyze whole table.
 	// So we don't need to check indexes.
-	if !exec.TableAnalyzed(tblStats) {
+	if tblStats.LastAnalyzeVersion == 0 {
```
Comment (hi-rustin) [inline pkg/statistics/handle/autoanalyze/refresher/refresher.go:442]: ditto

**Review by hi-rustin — COMMENTED** [review]
rest LGTM

Thanks for working on this! 👍

**Review by hi-rustin — APPROVED** [review]
except https://github.com/pingcap/tidb/pull/51489/files#r1517435884

Looks good to me. Thanks! 👍

**Review by hawkingrei — APPROVED** [review]

**Pull request merged**